docs/source/python/integration/python_r.rst

```diff
@@ -0,0 +1,312 @@
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+Integrating PyArrow with R
+==========================
+
+Arrow supports exchanging data within the same process through the
+:ref:`c-data-interface`.
+
+This can be used to exchange data between Python and R functions and
+methods so that the two languages can interact without any cost of
+marshaling and unmarshaling data.
+
+.. note::
+
+    The article takes for granted that you have a ``Python`` environment
+    with ``pyarrow`` correctly installed and an ``R`` environment with
+    ``arrow`` library correctly installed.
+
+Invoking R functions from Python
+--------------------------------
+
+Suppose we have a simple R function receiving an Arrow Array to
+add ``3`` to all its elements:
+
+.. code-block:: R
+
+    library(arrow)
+
+    addthree <- function(arr) {
+        return(arr + 3)
+    }
+
+We could save such function in a ``addthree.R`` file so that we can
+make it available for reuse.
+
+Once the ``addthree.R`` is created we can invoke any of its functions
+from Python using the 
+`rpy2 <https://rpy2.github.io/doc/latest/html/index.html>`_ library which
+enables a R runtime within the Python interpreter.
+
+``rpy2`` can be installed using ``pip`` like most python libraries
+
+.. code-block:: bash
+
+    $ pip install rpy2
+
+The most basic thing we can do with our ``addthree`` function is to
+invoke it from Python with a number and see how it will return the result.
+
+To do so we can create an ``addthree.py`` file which uses ``rpy2`` to
+import the ``addthree`` function from ``addthree.R`` file and invoke it:
+
+.. code-block:: python
+
+    import rpy2.robjects as robjects
+
+    # Load the addthree.R file
+    r_source = robjects.r["source"]
+    r_source("addthree.R")
+
+    # Get a reference to the addthree function
+    addthree = robjects.r["addthree"]
+
+    # Invoke the function
+    r = addthree(3)
+
+    # Access the returned value
+    value = r[0]
+    print(value)
+
+Running the ``addthree.py`` file will show how our Python code is able
+to access the ``R`` function and print the expected result:
+
+.. code-block:: bash
+
+    $ python addthree.py 
+    6.0
+
+If instead of passing around basic data types we want to pass around
+Arrow Arrays, we can do so relying on the ``rpy2-arrow`` module which
+implements rpy2 support for Arrow types.
+
+``rpy2`` can be installed through ``pip``:
+
+.. code-block:: bash
+
+    $ pip install rpy2-arrow
+
+``rpy2-arrow`` implements converters from pyarrow objects to r arrow objects,
```

Comment:
Capitalization: "PyArrow", "R", "Arrow". `arrow` and `pyarrow` should only be used when referring to the actual C++ (resp. Python) namespace.

Reply:
:+1: